src/turret.cpp

```diff
@@ -544,28 +544,30 @@ void vehicle::turrets_set_mode()
     }
 }
 
-npc vehicle::get_targeting_npc( const vehicle_part &pt ) const
+npc &vehicle::get_targeting_npc( const vehicle_part &pt ) const
 {
     // Make a fake NPC to represent the targeting system
-    npc cpu;
-    cpu.set_body();
-    cpu.set_fake( true );
+    static npc cpu;
```

Comment:
This might cause issues if the calling code tries to get targeting NPCs for two vehicles/parts at the same time. It might be better to cache the NPC in the vehicle part instead. Or, alternatively, return a `shared_ptr` of the NPC and return a newly constructed NPC if the previous pointer is still in use by outside code, if we want to save memory.

Reply:
There are only two usages of `vehicle::get_targeting_npc()`, both of them are safe ( the turret is fired immediately following
the request of cpu ). The only problem l can imagine is when someone wants concurrency, calling two cpus in parallel.

Reply:
The potential of misusing the function is what I'm worried about. Updating the function documentation or name to reflect that it is returning a reference to a single npc might help too, although I'd prefer changing the name because who reads the documentation.

Reply:
`class npc` has an explicitly deleted `operator=()` function, making it a hassle when wanting to store an instance of NPC inside a vehicle ( maybe also vp ).

Reply:
You can create a `targeting_npc` class that stores a `unique_ptr` to an NPC and resets the cached pointer on copy-assignment and copy-construction, so it can create a different NPC after being copied into another vehicle part.

Reply:
Thanks, this is really inspiring to me. I'll see if giving every turret a dedicated NPC still has better performance than creating a new one every time.

Reply:
Wow, it's actually better than using a static one.